src/microceph.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+"""Handle Ceph commands."""
+
+import logging
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+def add_osd_cmd(spec: str) -> None:
+    """Executes MicroCeph add osd cmd with provided spec."""
+    cmd = ["sudo", "microceph", "disk", "add", spec]
+    try:
+        logger.debug(f'Running command {" ".join(cmd)}')
+        process = subprocess.run(cmd, capture_output=True, text=True, check=True, timeout=180)
+        logger.debug(f"Command finished. stdout={process.stdout}, " f"stderr={process.stderr}")
+    except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as e:
+        logger.warning(e.stderr)
+        error_disk_already_exists = (
+            "Failed adding new disk: Failed to record disk: Failed to create "
+            '"disks" entry: UNIQUE constraint failed: disks.member_id, disks.path'
+        )
+        if error_disk_already_exists not in e.stderr:
```

Comment:
If the disk is present, the user should at least be notified. On the other hand, why is it that this isn't considered an error, anyway? (I know this code was present before, but if it needs changing, we should do it).

Reply:
ack, have removed the section where catch and add logic, the exception now rises directly to charm.py caller.